gematria/datasets/python/annotating_importer.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/datasets/annotating_importer.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <vector>
+
+#include "gematria/llvm/canonicalizer.h"
+#include "llvm/Support/Error.h"
+#include "pybind11/cast.h"
+#include "pybind11/detail/common.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+#include "pybind11_abseil/import_status_module.h"
+#include "pybind11_abseil/status_casters.h"
+#include "pybind11_protobuf/native_proto_caster.h"
+
+namespace gematria {
+
+namespace py = ::pybind11;
+
+PYBIND11_MODULE(annotating_importer, m) {
+  m.doc() = "Support code for importing annotated basic block data.";
+
+  py::google::ImportStatusModule();
+
+  py::class_<AnnotatingImporter>(m, "AnnotatingImporter")
+      .def(  //
+          py::init<const Canonicalizer* /* canonicalizer */>(),
```

Comment:
We might need a [`keep_alive` policy](https://pybind11.readthedocs.io/en/stable/advanced/functions.html#keep-alive) here.